android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/MapsFragment.kt

```diff
@@ -51,27 +68,65 @@ class MapsFragment : Fragment() {
         }
     }
 
+    private fun initPlaces() {
+        context?.let { Places.initialize(it, getString(R.string.google_maps_key)) }
+        placesClient = context?.let { Places.createClient(it) }!!
+    }
+
+    private fun setPlacesSearchBias() {
+        val autocompleteFragment = childFragmentManager.findFragmentById(R.id.autocomplete_fragment) as AutocompleteSupportFragment
+        // Search nearby result
+        currentLatLng?.let{
+            Log.i(TAG, "currentLatLng")
+            autocompleteFragment.setLocationBias(
+                RectangularBounds.newInstance(
+                    LatLng(currentLatLng!!.latitude - 1, currentLatLng!!.longitude - 1),
+                    LatLng(currentLatLng!!.latitude + 1, currentLatLng!!.longitude + 1)
+                ))
+        }
+    }
+
     private fun setUpMap() {
         if (checkSelfPermission(context as Activity,
                 Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
             requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST_CODE)
             return
         }
 
+        // Add and adjust the position of MyLocation button.
         map.isMyLocationEnabled = true
+        map.setPadding(0,(1.5 * autocompleteLayout.height).toInt(),0,0)
 
         fusedLocationClient.lastLocation.addOnSuccessListener(this.activity as Activity) { location ->
             // Got last known location. In some rare situations this can be null.
             if (location != null) {
                 lastLocation = location
-                val currentLatLng = LatLng(location.latitude, location.longitude)
-                map.animateCamera(CameraUpdateFactory.newLatLngZoom(currentLatLng, 12f))
-                map.addMarker(MarkerOptions().position(currentLatLng).title("My location"))
-                map.moveCamera(CameraUpdateFactory.newLatLng(currentLatLng))
+                currentLatLng = LatLng(location.latitude, location.longitude)
+                map.animateCamera(CameraUpdateFactory.newLatLngZoom(currentLatLng, 14f))
```

Comment:
Try to avoid magic numbers like `14f`. Define a constant instead and replace it here. Same applies to other magic numbers in file as well.